cmd/gpu_fakedev/gpu_fakedev.go

```diff
@@ -291,6 +296,88 @@ func generateDriFiles(opts genOptions) {
 	log.Printf("Done, created %d dirs, %d devices and %d files.", opts.dirs, opts.devs, opts.files)
 }
 
+func makeXelinkSideCar(topology string, gpus, tiles int, connections string) {
+	if topology != fullyConnected {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d)", gpus, tiles)
+	} else {
+		log.Printf("XELINK: generate xelink sidecar label file, using (GPUs: %d, Tiles: %d, Topology: %s)", gpus, tiles, topology)
+	}
+
+	if topology == fullyConnected {
+		saveSideCarFile(buildConnectionList(gpus, tiles))
+	} else {
+		saveSideCarFile(connections)
+	}
+}
+
+func buildConnectionList(gpus, tiles int) string {
+	mm := 0
+
+	var nodes = make([]string, 0)
+
+	for mm < gpus {
+		nn := 0
+		for nn < tiles {
+			nodes = append(nodes, fmt.Sprintf("%d.%d", mm, nn))
+			nn++
+		}
+		mm++
```

Comment:
Initialization and increments are normally done in the `for` line:
```suggestion
	var nodes = make([]string, 0)

	for mm := 0; mm < gpus; mm++ {
		for nn := 0; nn < tiles; nn++ {
			nodes = append(nodes, fmt.Sprintf("%d.%d", mm, nn))
		}
```